tests failing on test_07_l2vpn_return_codes.py

the following tests are failing on the daily execution:
```
_________ TestE2EReturnCodesEditL2vpn.test_010_edit_l2vpn_vlan_code201 _________

self = <test_07_l2vpn_return_codes.TestE2EReturnCodesEditL2vpn object at 0x7faa7a29f250>

   def test_010_edit_l2vpn_vlan_code201(self):
       """
       Test the return code for creating a SDX L2VPN
       201: L2VPN Service Modified
       Edit vlan
       """
       api_url = SDX_CONTROLLER + '/l2vpn/1.0'
     self.payload['endpoints'][0]['vlan'] = "200"
E       AttributeError: 'TestE2EReturnCodesEditL2vpn' object has no attribute 'payload'

tests/test_07_l2vpn_return_codes.py:64: AttributeError
_______ TestE2EReturnCodesEditL2vpn.test_011_edit_l2vpn_port_id_code201 ________

self = <test_07_l2vpn_return_codes.TestE2EReturnCodesEditL2vpn object at 0x7faa7aab4a90>

   def test_011_edit_l2vpn_port_id_code201(self):
       """
       Test the return code for creating a SDX L2VPN
       201: L2VPN Service Modified
       Edit port_id
       """
       api_url = SDX_CONTROLLER + '/l2vpn/1.0'
     self.payload['endpoints'][0]['port_id'] = "urn:sdx:port:[sax.net](http://sax.net/):Sax01:50"
E       AttributeError: 'TestE2EReturnCodesEditL2vpn' object has no attribute 'payload'

tests/test_07_l2vpn_return_codes.py:76: AttributeError
__ TestE2EReturnCodesEditL2vpn.test_020_edit_l2vpn_with_vlan_integer_code400 ___

self = <test_07_l2vpn_return_codes.TestE2EReturnCodesEditL2vpn object at 0x7faa7a1d4a10>

   def test_020_edit_l2vpn_with_vlan_integer_code400(self):
       """
       Test the return code for creating a SDX L2VPN
       400: Request does not have a valid JSON or body is incomplete/incorrect
       -> Wrong vlan: vlan is not a string
       """
       api_url = SDX_CONTROLLER + '/l2vpn/1.0'
     self.payload['endpoints'][0]['vlan'] = 300
E       AttributeError: 'TestE2EReturnCodesEditL2vpn' object has no attribute 'payload'

tests/test_07_l2vpn_return_codes.py:87: AttributeError
_ TestE2EReturnCodesEditL2vpn.test_021_edit_l2vpn_with_vlan_out_of_range_code400 _

self = <test_07_l2vpn_return_codes.TestE2EReturnCodesEditL2vpn object at 0x7faa79ff7750>

   def test_021_edit_l2vpn_with_vlan_out_of_range_code400(self):
       """
       Test the return code for creating a SDX L2VPN
       400: Request does not have a valid JSON or body is incomplete/incorrect
       -> Wrong vlan: vlan is out of range 1-4095
       """
       api_url = SDX_CONTROLLER + '/l2vpn/1.0'
     self.payload['endpoints'][0]['vlan'] = 5000
E       AttributeError: 'TestE2EReturnCodesEditL2vpn' object has no attribute 'payload'

tests/test_07_l2vpn_return_codes.py:98: AttributeError
____ TestE2EReturnCodesEditL2vpn.test_022_edit_l2vpn_with_vlan_all_code400 _____

self = <test_07_l2vpn_return_codes.TestE2EReturnCodesEditL2vpn object at 0x7faa79ff5b90>

   def test_022_edit_l2vpn_with_vlan_all_code400(self):
       """
       Test the return code for creating a SDX L2VPN
       400: Request does not have a valid JSON or body is incomplete/incorrect
       -> Wrong vlan: since one endpoint has the "all" option, all endpoints must have the same value
       """
       api_url = SDX_CONTROLLER + '/l2vpn/1.0'
     self.payload['endpoints'][0]['vlan'] = "all"
E       AttributeError: 'TestE2EReturnCodesEditL2vpn' object has no attribute 'payload'

tests/test_07_l2vpn_return_codes.py:109: AttributeError
_ TestE2EReturnCodesEditL2vpn.test_023_edit_l2vpn_with_body_incomplete_code400 _

self = <test_07_l2vpn_return_codes.TestE2EReturnCodesEditL2vpn object at 0x7faa79ff6310>

   def test_023_edit_l2vpn_with_body_incomplete_code400(self):
       """
       Test the return code for creating a SDX L2VPN
       400: Request does not have a valid JSON or body is incomplete/incorrect
       -> Body incomplete: vlan attribute is missing on an endpoint
       """
       api_url = SDX_CONTROLLER + '/l2vpn/1.0'
       payload = {
           "name": "Test L2VPN request",
           "endpoints": [
               {"port_id": "urn:sdx:port:[ampath.net](http://ampath.net/):Ampath3:50","vlan": "300"},
               {"port_id": "urn:sdx:port:[tenet.ac.za](http://tenet.ac.za/):Tenet03:50"}
           ]
       }
       response = requests.patch(f"{api_url}/{self.key}", json=payload)
     assert response.status_code == 400, response.text
E       AssertionError: "Failed, reason: Unknown connection request (ID: 4a8d04ab-1a59-4b19-9f26-3df739cfe32e)"
E         
E       assert 500 == 400
E        +  where 500 = <Response [500]>.status_code

tests/test_07_l2vpn_return_codes.py:129: AssertionError
_ TestE2EReturnCodesEditL2vpn.test_024_edit_l2vpn_with_body_incorrect_code400 __

self = <test_07_l2vpn_return_codes.TestE2EReturnCodesEditL2vpn object at 0x7faa79ff6cd0>

   def test_024_edit_l2vpn_with_body_incorrect_code400(self):
       """
       Test the return code for creating a SDX L2VPN
       400: Request does not have a valid JSON or body is incomplete/incorrect
       -> Body incorrect: port_id
       """
       api_url = SDX_CONTROLLER + '/l2vpn/1.0'
       payload = {
           "name": "Test L2VPN request",
           "endpoints": [
               {"port_id": "urn:sdx:port:[ampath.net](http://ampath.net/):Ampath","vlan": "300"},
               {"port_id": "urn:sdx:port:[tenet.ac.za](http://tenet.ac.za/):Tenet03:50","vlan": "300"}
           ]
       }
       response = requests.patch(f"{api_url}/{self.key}", json=payload)
     assert response.status_code == 400, response.text
E       AssertionError: "Failed, reason: Unknown connection request (ID: 4a8d04ab-1a59-4b19-9f26-3df739cfe32e)"
E         
E       assert 500 == 400
E        +  where 500 = <Response [500]>.status_code

tests/test_07_l2vpn_return_codes.py:146: AssertionError
_____ TestE2EReturnCodesEditL2vpn.test_040_edit_l2vpn_not_found_id_code404 _____

self = <test_07_l2vpn_return_codes.TestE2EReturnCodesEditL2vpn object at 0x7faa79ff77d0>

   def test_040_edit_l2vpn_not_found_id_code404(self):
       """
       Test the return code for creating a SDX L2VPN
       404: L2VPN Service ID not found
       """
       api_url = SDX_CONTROLLER + '/l2vpn/1.0'
       key = '11111111-1111-1111-1111-111111111111'
     response = requests.patch(f"{api_url}/{key}", json=self.payload)
E       AttributeError: 'TestE2EReturnCodesEditL2vpn' object has no attribute 'payload'

tests/test_07_l2vpn_return_codes.py:173: AttributeError
_______ TestE2EReturnCodesEditL2vpn.test_050_edit_l2vpn_conflict_code409 _______

self = <test_07_l2vpn_return_codes.TestE2EReturnCodesEditL2vpn object at 0x7faa79ff54d0>

   def test_050_edit_l2vpn_conflict_code409(self):
       """
       Test the return code for creating a SDX L2VPN
       409: Conflicts with a different L2VPN
       """
       api_url = SDX_CONTROLLER + '/l2vpn/1.0'
       # Create a new l2vpn with similar endpoints to the existing one. Only the vlan varies
       payload = {
           "name": "Test L2VPN request",
           "endpoints": [
               {"port_id": "urn:sdx:port:[ampath.net](http://ampath.net/):Ampath2:50","vlan": "500"},
               {"port_id": "urn:sdx:port:[sax.net](http://sax.net/):Sax01:50","vlan": "500"}
           ]
       }
       response = requests.post(api_url, json=payload)
       assert response.status_code == 201, response.text

       # edit the first l2pvn to match the newly created one
     self.payload['endpoints'][0]['vlan'] = "500"
E       AttributeError: 'TestE2EReturnCodesEditL2vpn' object has no attribute 'payload'

tests/test_07_l2vpn_return_codes.py:194: AttributeError
_ TestE2EReturnCodesEditL2vpn.test_060_edit_l2vpn_with_min_bw_out_of_range_code410 _

self = <test_07_l2vpn_return_codes.TestE2EReturnCodesEditL2vpn object at 0x7faa79ff4a50>

   def test_060_edit_l2vpn_with_min_bw_out_of_range_code410(self):
       """
       Test the return code for creating a SDX L2VPN
       410: Can't fulfill the strict QoS requirements
       Case: min_bw out of range (value must be in [0-100])
       """
       api_url = SDX_CONTROLLER + '/l2vpn/1.0'
     self.payload['qos_metrics'] = {'min_bw':{'value': 101}}
E       AttributeError: 'TestE2EReturnCodesEditL2vpn' object has no attribute 'payload'

tests/test_07_l2vpn_return_codes.py:206: AttributeError
_ TestE2EReturnCodesEditL2vpn.test_061_edit_l2vpn_with_max_delay_out_of_range_code410 _

self = <test_07_l2vpn_return_codes.TestE2EReturnCodesEditL2vpn object at 0x7faa7ab89810>

   def test_061_edit_l2vpn_with_max_delay_out_of_range_code410(self):
       """
       Test the return code for creating a SDX L2VPN
       410: Can't fulfill the strict QoS requirements
       Case: max_delay out of range (value must be in [0-1000])
       """
       api_url = SDX_CONTROLLER + '/l2vpn/1.0'
     self.payload['qos_metrics'] = {'max_delay':{'value': 1001}}
E       AttributeError: 'TestE2EReturnCodesEditL2vpn' object has no attribute 'payload'

tests/test_07_l2vpn_return_codes.py:217: AttributeError
_ TestE2EReturnCodesEditL2vpn.test_062_edit_l2vpn_with_max_number_oxps_out_of_range_code410 _

self = <test_07_l2vpn_return_codes.TestE2EReturnCodesEditL2vpn object at 0x7faa7ab948d0>

   def test_062_edit_l2vpn_with_max_number_oxps_out_of_range_code410(self):
       """
       Test the return code for creating a SDX L2VPN
       410: Can't fulfill the strict QoS requirements
       Case: max_number_oxps out of range (value must be in [0-100])
       """
       api_url = SDX_CONTROLLER + '/l2vpn/1.0'
     self.payload['qos_metrics'] = {'max_number_oxps':{'value': 101}}
E       AttributeError: 'TestE2EReturnCodesEditL2vpn' object has no attribute 'payload'

tests/test_07_l2vpn_return_codes.py:228: AttributeError
```

--- a/docker-compose.yml
+++ b/docker-compose.yml
@@ -185,6 +185,8 @@ services:
       - ampath
       - sax
       - tenet
+    environment:
+      PYTHONDONTWRITEBYTECODE: "1"
     entrypoint:
       - "/bin/bash"
       - "-x"

--- a/tests/test_07_l2vpn_return_codes.py
+++ b/tests/test_07_l2vpn_return_codes.py
@@ -37,22 +37,21 @@ class TestE2EReturnCodesEditL2vpn:
         response = requests.get(api_url)
         assert response.status_code == 200, response.text
         response_json = response.json()
+        cls.payload = {
+            "name": "Test L2VPN request",
+            "endpoints": [
+                {"port_id": "urn:sdx:port:ampath.net:Ampath2:50","vlan": "50"},
+                {"port_id": "urn:sdx:port:tenet.ac.za:Tenet03:50","vlan": "50"}
+            ]
+        }
         if len(response_json) == 0:
             # Create an L2VPN to edit later
             api_url = SDX_CONTROLLER + '/l2vpn/1.0'
-            cls.payload = {
-                "name": "Test L2VPN request",
-                "endpoints": [
-                    {"port_id": "urn:sdx:port:ampath.net:Ampath2:50","vlan": "50"},
-                    {"port_id": "urn:sdx:port:tenet.ac.za:Tenet03:50","vlan": "50"}
-                ]
-            }
             response = requests.post(api_url, json=cls.payload)
             assert response.status_code == 201, response.text
-
-        response = requests.get(api_url)
-        data = response.json()
-        cls.key = list(data.keys())[0]
+            cls.key = response.json()["service_id"]
+        else:
+            cls.key = list(response_json.keys())[0]
 
     def test_010_edit_l2vpn_vlan_code201(self):
         """

